### PR TITLE
Convert react charts demos to functionality 

### PR DESCRIPTION
**Pull request by dxvladislavvolkov**

### CODE REVIEW

**Review by alexanderPolosatov — DISMISSED** [review]

File: JSDemos/Demos/Charts/BarSparklines/React/App.js [hunk 1]
```diff
@@ -20,57 +19,53 @@ const dataSource = new DataSource({
...
 });
 
-class App extends React.Component {
-  constructor(props) {
-    super(props);
-    this.source = dataSource;
-    this.onValueChanged = this.onValueChanged.bind(this);
-  }
+function App() {
+  React.useEffect(() => {
```
Comment (ivanblinov2k17) [inline JSDemos/Demos/Charts/BarSparklines/React/App.js]: why do we need to do this if it's inside onvalueChange callback?

Comment (dxvladislavvolkov) [reply JSDemos/Demos/Charts/BarSparklines/React/App.js]: Yes, it is strange. I will fix it

File: JSDemos/Demos/Charts/ExportCustomMarkup/React/App.js [hunk 2]
```diff
@@ -15,77 +15,69 @@ function prepareMarkup(chartSvg, markup) {
...
+    setChildRef(ref);
+  }, []);
 
-  onRef(arg) {
-    this.child = arg;
-  }
+  return (
+    <div id="chart-demo">
+      <div className="chart_environment">
+        <Form onRef={onRef} />
```
Comment (ivanblinov2k17) [inline JSDemos/Demos/Charts/ExportCustomMarkup/React/App.js]: Whats this for? i wasnt able to find onRef in our docs
its a bit strange that we store ref in state and set it this way

Comment (dxvladislavvolkov) [reply JSDemos/Demos/Charts/ExportCustomMarkup/React/App.js]: It is custom form component

Comment (ivanblinov2k17) [reply JSDemos/Demos/Charts/ExportCustomMarkup/React/App.js]: ok, found out that is a prop for custom form component
maybe do the same with forwardRef?

File: JSDemos/Demos/Charts/ParetoChart/React/App.js [hunk 3]
```diff
@@ -115,4 +31,86 @@ function customizePercentageText({ valueText }) {
   return `${valueText}%`;
 }
 
+function App() {
+  const [dataArray, setDataArray] = React.useState([]);
+
+  React.useEffect(() => {
+    const data = complaintsData.sort((a, b) => b.count - a.count);
```
Comment (ivanblinov2k17) [inline JSDemos/Demos/Charts/ParetoChart/React/App.js]: we dont need state and useeffect here, all can be moved outside component

**Review by ivanblinov2k17 — APPROVED** [review]

**Pull request merged**